### PR TITLE
Analysis: Tech/787/enable coverage report

### LINKED ISSUE #787: CodeCharta Analysis Coverage is 0%
# Bug

## Expected Behavior

Coverage fulfills the quality profile we set ourselves.

## Actual Behavior

* Version 1.22.0 March 2019 had 48% coverage
* Version 1.23.0 April 2019 dropped to 12%, then to "up" to 14.1%
* Version 1.24.0 May 2019 went from 14.1% to 0%

### PR DESCRIPTION
**Pull request by NearW**
# Analysis publishes coverage results to sonar

closes #787

## Description

As already described in the issue, you have to generate coverage reports using jacoco and add them to the sonarqube task in gradle. The tricky part here is, that jacoco generates multiple coverage reports for each sub-project. You need to merge them together. I found a working solution online and integrated it.

You can test this locally by running `./gradlew codeCoverageReport` and check the build folder. We'll see if this works on Travis when we merged this branch, since only then the sonar stage is triggered. 

## Definition of Done

A task/pull request will not be considered to be complete until all these items can be checked off.

- [ ] **All** requirements mentioned in the issue are implemented
- [ ] Does match the Code of Conduct and the Contribution file
- [ ] Task has its own **GitHub issue** (something it is solving)
  - [ ] Issue number is **included in the commit messages** for traceability
- [ ] **Update the README.md** with any changes/additions made
- [ ] **Update the CHANGELOG.md** with any changes/additions made
- [ ] **Enough test coverage to ensure that uncovered changes do not break functionality**
- [ ] **All tests pass**
- [ ] **Descriptive pull request text**, answering:
  - What problem/issue are you fixing?
  - What does this PR implement and how?
- [ ] **Assign your PR to someone** for a code review
  - This person _will be contacted **first**_ if a bug is introduced into `master`
- [ ] **Manual testing** did not fail

### CODE REVIEW

**Comment by NearW** [comment]
> Isn't sonar failing on travis in general right now? I guess that has to be fixed first?

That is true. There is an authorization issue, which is weird, since both projects should require the same login data and the visualization is working correctly.

Nontheless, I don't see how we should ever get a coverage on sonarcloud without these settings.

**Comment by NearW** [comment]
@BridgeAR I invested some more time into this and removed the SonarQube gradle plugin and instead used to sonar-scanner as we do with the visualization. This requires a sonar-properties-file in which we can configure our ruleset, which is actually nice to have.

The biggest conclusion here is that since we changed our SONAR_TOKEN in Travis, we can't publish our results to visualization anyone as well. In order to do that, we have to give `execute analysis` privileges in sonarcloud. This can only be done by the organization. Everything should be working then.

Let's get in contact with Stefan and resolve this and then look at this PR again.

**Comment by BridgeAR** [comment]
I reached out to Stefan. Let's see what he says.

File: .travis.yml [hunk 1]
```diff
@@ -84,10 +84,12 @@ jobs:
 
     - name: "Publish Analysis Sonar Results"
       before_script: cd analysis
-      script: ./gradlew build integrationTest
+      script:
+        - ./gradlew build test integrationTest
```
Comment (NearW) [inline .travis.yml:88]: Remove test here since it is a pre-requisite of `jacocoTestReport`

**Pull request merged**